pkg/agent/multicluster/pod_route_controller.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicluster
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	mcv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	mcclientset "antrea.io/antrea/multicluster/pkg/client/clientset/versioned"
+	mclisters "antrea.io/antrea/multicluster/pkg/client/listers/multicluster/v1alpha1"
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/openflow"
+	"antrea.io/antrea/pkg/ovs/ovsconfig"
+)
+
+const (
+	// Default number of workers processing a resource change
+	defaultWorkerNum = 5
+)
+
+// MulticlusterNodeRouteController generates necessary L3 forwarding rules
+// for any Pod to forward cross-cluster traffic from MC Gateway to regular Nodes
```

Comment:
This sentence is too long to read. How about:
> MCPodRouteController generates L3 forwarding flows to forward cross-cluster traffic from MC Gateway to Pods on other Nodes inside a member cluster. It is required when networkPolicyOnly, noEncap or hybrid mode are configured, to forward the traffic through tunnels between Gateway and other Nodes, as otherwise the traffic will not go through tunnels in those modes.